libr/cons/dietline.c

```diff
@@ -442,7 +442,114 @@ R_API int r_line_hist_chop(const char *file, int limit) {
 	return 0;
 }
 
+static void draw_selection_widget() {
+	RCons *cons = r_cons_singleton ();
+	RSelWidget *sel_widget = I.sel_widget;
+	int y, pos_y = cons->rows, pos_x = r_str_ansi_len (I.prompt);
+
+	for (y = 0; y < sel_widget->options_len; y++) {
+		sel_widget->w = R_MAX (sel_widget->w, strlen (sel_widget->options[y]));
+	}
+	sel_widget->w = R_MIN (sel_widget->w, R_SELWIDGET_MAXW);
+
+	char *background_color = cons->color ? cons->pal.widget_bg : Color_INVERT;
+	char *selected_color = cons->color ? cons->pal.widget_sel : Color_INVERT_RESET;
+	bool scrollbar = sel_widget->options_len > R_SELWIDGET_MAXH;
+	int scrollbar_y = 0, scrollbar_l = 0;
+	if (scrollbar) {
+		scrollbar_y = (R_SELWIDGET_MAXH * (sel_widget->selection - sel_widget->scroll)) / sel_widget->options_len;
+		scrollbar_l = (R_SELWIDGET_MAXH * R_SELWIDGET_MAXH) / sel_widget->options_len;
+	}
+
+	for (y = 0; y < R_MIN (sel_widget->h, R_SELWIDGET_MAXH); y++) {
+		r_cons_gotoxy (pos_x + 1, pos_y - y - 1);
+		int scroll = R_MAX (0, sel_widget->selection - sel_widget->scroll);
+		char *option = y < sel_widget->options_len ? sel_widget->options[y + scroll] : "";
+		r_cons_printf ("%s", sel_widget->selection == y + scroll ? selected_color : background_color);
+		r_cons_printf ("%-*.*s", sel_widget->w, sel_widget->w, option);
+		if (scrollbar) {
+			r_cons_strcat (background_color);
+			r_cons_printf ("%s", R_BETWEEN (scrollbar_y, y, scrollbar_y + scrollbar_l) ? SELWIDGET_SCROLLCHAR : " ");
```

Comment:
SELWIDGET_SCROLLCHAR is utf8, you must set a var at the begining of the function to be that one or an ascii alternative when utf8 is not set